tools/bumper/bumper.go

```diff
@@ -0,0 +1,139 @@
+	components, err := parseComponentsYaml(inputArgs.componentsConfigPath)
+	if err != nil {
+		exitWithError(errors.Wrap(err, "Failed to parse components yaml"))
+	}
+
+	for componentName, component := range components {
+		Logger.Printf("Checking if %s needs bumping", componentName)
+
+		err = printCurrentComponentParams(component)
+		if err != nil {
+			exitWithError(errors.Wrapf(err, "Failed to print component %s", componentName))
+		}
+
+		currentReleaseVersion, err := getCurrentReleaseTag(component.Url)
+		if err != nil {
+			exitWithError(errors.Wrapf(err, "Failed to get latest release version tag from %s", component.Url))
+		}
+
+		latestReleaseVersion, latestReleaseCommit, err := getLatestReleaseInfo(component.Url)
+		if err != nil {
+			exitWithError(errors.Wrapf(err, "Failed to get latest release version tag from %s", componentName))
+		}
+
+		//TODO updatePolicyDummy is a placeholder. In future task will be replaced by proper parameter set in the configYaml
+		updatePolicyDummy := "static"
+		bumpNeeded, err := isBumpNeeded(currentReleaseVersion, latestReleaseVersion, updatePolicyDummy)
+		if err != nil {
+			exitWithError(errors.Wrapf(err, "Failed to discover if Bump need for %s", componentName))
+		}
+
+		if bumpNeeded {
+			Logger.Printf("Bumping %s from %s to %s", componentName, currentReleaseVersion, latestReleaseVersion)
+			//reset --hard git repo
+
+			//PR name
+			PRTitle := fmt.Sprintf("Bump %s to %s", componentName, latestReleaseVersion)
+			Logger.Printf("PR title %s", PRTitle)
+			//Create PR
+
+			//update component's entry in config yaml
+			component.Commit = latestReleaseCommit
+			component.Metadata.Tag = latestReleaseVersion
+			err = updateComponentsYaml(inputArgs.componentsConfigPath, components)
+			if err != nil {
+				exitWithError(errors.Wrap(err, "Failed to update components yaml"))
+			}
+
+			//run bump script
+			cmd := exec.Command("make", fmt.Sprintf("bump-%s", componentName))
+			if out, err := cmd.CombinedOutput(); err != nil {
+				exitWithError(errors.Wrapf(err, "Failed to run bump script. StdOut = %s", string(out)))
+			}
+
+			//create a new branch name
+			BranchName := strings.Replace(PRTitle, " ", "_", -1)
+			Logger.Printf("Opening new Branch %s", BranchName)
+
+			//push branch to PR
+
+		}
+	}
+}
+
+func getLatestReleaseInfo(componentUrl string) (string, string, error) {
+	return "", "", nil
+}
+
+func getCurrentReleaseTag(componentUrl string) (string, error) {
+	return "", nil
+}
+
+func isBumpNeeded(currentReleaseVersion, latestReleaseVersion, updatePolicy string) (bool, error) {
+	Logger.Printf("currentReleaseVersion: %s, latestReleaseVersion: %s, updatePolicy: %s\n", currentReleaseVersion, latestReleaseVersion, updatePolicy)
+	return false, nil
+}
+
+func initBumper(inputParams *inputParams) (*log.Logger) {
+	Logger = initLog()
+	// cold start message (visibly unique in log)
+	Logger.Printf("~~~~~~~~Init Bumper Script~~~~~~~~")
+	initFlags(inputParams)
+
+	return Logger
+}
```

Comment:
As say before `initBumper` means nothing, do the logger init at main directly and call initFlags after it, also this functions has as mix of responsabilities that opens the door to turn into a pandora box.

Reply:
removed `initBumper`
Done